client/data/marketplace/use-wpcom-plugins-query.ts

```diff
@@ -120,26 +124,24 @@ export const useWPCOMPlugin = (
 export const useWPCOMPlugins = ( slugs: Array< string > ): Array< UseQueryResult< any > > => {
 	return useQueries( {
 		queries: slugs.map( ( slug ) => {
-			const [ cacheKey, fetchFn ] = getWPCOMPluginQueryParams( slug );
-
-			return {
-				queryKey: cacheKey,
-				queryFn: fetchFn,
-			};
+			return getWPCOMPluginQueryParams( slug );
 		} ),
 	} );
 };
 
-export const getWPCOMFeaturedPluginsQueryParams = (): [ QueryKey, QueryFunction ] => {
-	const cacheKey = [ 'plugins-featured-list-normalized' ];
-	const fetchFn = () =>
+export const getWPCOMFeaturedPluginsQueryParams = (): {
+	queryKey: QueryKey;
+	queryFn: QueryFunction;
+} => {
+	const queryKey = [ 'plugins-featured-list-normalized' ];
+	const queryFn = () =>
 		wpcom.req
 			.get( {
 				path: featuredPluginsApiBase,
 				apiNamespace: pluginsApiNamespace,
 			} )
 			.then( normalizePluginsList );
-	return [ cacheKey, fetchFn ];
+	return { queryKey, queryFn };
```

Comment:
Noting that `getWPCOMFeaturedPluginsQueryParams ` is used here:

https://github.com/Automattic/wp-calypso/blob/8f1fa912356c59738ed483a536f84bb85f191ca1/client/my-sites/plugins/controller-logged-out.js#L61

and we might need to do some changes to accommodate the function signature difference.